File: typescript/bob/bob.ts
class Bob {
    hey(content: string): string {
        const isShouting = (s: string) => s.search(/[A-Z]/g) >= 0 && s === s.toUpperCase()
        const isAsking = (s: string) => s.endsWith('?')
        const isSilent = (s: string) => s.trim() === ''

        if (isShouting(content)) {
            return 'Whoa, chill out!'
        } else if (isAsking(content)) {
            return 'Sure.'
        } else if (isSilent(content)) {
            return 'Fine. Be that way!'
        } else {
            return 'Whatever.'
        }
    }

}

export default Bob

File: go/bob/bob.go
// Package bob implements bob's responses
package bob

import "strings"

type saying string

const alphabet = "abcdefghijklmnopqrstuvwxyz"

// Hey returns response to given remark
func Hey(remark string) string {
	switch {
	case isQuestion(remark) && isShout(remark):
		return "Calm down, I know what I'm doing!"
	case isQuestion(remark):
		return "Sure."
	case isShout(remark):
		return "Whoa, chill out!"
	case isSilent(remark):
		return "Fine. Be that way!"
	default:
		return "Whatever."
	}
}

func isQuestion(s string) bool {
	return strings.HasSuffix(strings.TrimSpace(s), "?")
}

func isShout(s string) bool {
	return strings.ContainsAny(alphabet, strings.ToLower(s)) && strings.ToUpper(s) == s
}

func isSilent(s string) bool {
	return strings.TrimSpace(s) == ""
}
